src/excmds.ts

```diff
@@ -2877,6 +2897,26 @@ async function getclip() {
     }
 }
 
+/**
+ * Fetches primary selection only
+ *
+ * @hidden
+ */
+//#background_helper
+async function getprimsel() {
```

Comment:
I feel like this error handling should potentially be in Native.clipboard - what do you think, @glacambre? Were you trying to avoid round trips?

Reply:
>I feel like this error handling should potentially be in Native.clipboard

I remember first attempting to do that and then going back to throwing errors because I couldn't decide what Native.clipboard should return on failure.
- It obviously can't return a string because we wouldn't be able to tell the difference between an error and the content of the clipboard.
- It can't return undefined or null because then the user wouldn't know what went wrong (did the failure happen because I don't have the native messenger? Or because I don't have xsel/xclip installed? Or because the call to xsel/xclip failed?).
So we're left with either enums/numbers or custom monads.
- Enums/numbers are not very ergonomic IMO because they force you to either write error messages yourself or to fetch them from a table stored somewhere else.
- I didn't see a point in custom monads because promises already are monads.

Is there another way I haven't thought of?

Reply:
Why not an exception?

Reply:
Because the code already throws exceptions so I thought bovine3dom wanted Native.clipboard() not to throw. Did I misunderstand something?